src/project/settings.rs

```diff
@@ -65,17 +65,31 @@ impl ProjectSettings {
     pub fn get_trunk(&mut self) -> Result<String> {
         let trunk = match self.trunk.to_owned() {
             Some(a) => a,
-            None => ask_for_trunk()?,
-        };
+            None => {
+                let trunk = ask_for_trunk()?;
+                self.set_trunk(&trunk)?;
 
-        self.set_trunk(&trunk)?;
+                trunk
+            }
+        };
 
         Ok(trunk)
     }
 
     fn save(&self) -> Result<()> {
         let content = serde_json::to_string(self)?;
-        std::fs::write(".git/.gi_project_config", content).map_err(|_| {
+
+        let Some(repository_root) = get_git_client()?.get_repository_root() else {
+            return Err(eyre!("You are not inside a git repository.").suggestion(
+                "Run `gi` inside a git repository or run `git init` to create a new one.",
+            ));
+        };
+
+        std::fs::write(
+            Path::new(&repository_root).join(".git/gi_project_config"),
+            content,
+        )
+        .map_err(|_| {
             eyre!("Failed to save project settings")
                 .suggestion("Check if you have write permissions to the .git directory.")
         })?;
```

Comment:
```suggestion
        .context("Failed to save project settings").suggestion("...")?;
```

Can't you write something like this instead here ?

Reply:
Done. Thanks!